presto-raptor/src/main/java/com/facebook/presto/raptor/storage/OrcStorageManager.java

```diff
@@ -215,18 +219,22 @@ public ConnectorPageSource getPageSource(
             ImmutableList.Builder<Integer> columnIndexes = ImmutableList.builder();
             for (int i = 0; i < columnIds.size(); i++) {
                 long columnId = columnIds.get(i);
+                if (isHiddenColumn(columnId)) {
+                    columnIndexes.add(addSpecialIndex(columnId));
+                    continue;
+                }
                 if (isShardRowIdColumn(columnId)) {
```

Comment:
Should `ROWID_COLUMN` and `SHARD_UUID_COLUMN` be removed here, since they are handled above?